src/libopenrave-core/colladaparser/colladareader.cpp

```diff
@@ -898,6 +903,11 @@ class ColladaReader : public daeErrorHandler
             }
         }
 
+        if( bSuccess ) {
+            // for extracting a single body from file, need to assign a random body id to not conflict with what's already in environment
+            probot->_id = str(boost::format("body%d")%utils::GetNanoTime());
```

Comment:
@ziyan random values like this are not a good idea. Please change the implementation so the same consistent value is produced loading the same scene, no matter when or how many times i load that scene.

Reply:
For now, I think empty id is okay for this case. Resolved.